wgengine/magicsock/magicsock.go

```diff
@@ -69,6 +69,14 @@ const (
 	// _linux variant.
 	discoMagic1 = 0x5453f09f
 	discoMagic2 = 0x92ac
+
+	// UDP socket read/write buffer size (12MB). This will be set using
+	// SO_RECVBUF/SO_SNDBUF, and is often clamped by the OS to a much smaller
+	// value. This buffer includes the cost of packet headers and OS internal
+	// management structures (i.e. skb's on Linux). Linux has a special
+	// implementation to use SO_RCVBUFFORCE/SO_SNDBUFFORCE to bypass the clamp -
+	// see magicksock_linux.go.
+	socketBufferSize = 12 << 20
```

Comment:
Why 12M? It's not a power of 2, which makes me think it was selected somehow. It'd be nice to have some breadcrumbs for future us to remember how this was arrived at (or explicitly state that it's an arbitrary high value, if that's what it is)

Reply:
It is somewhat arbitrary, but I just set it down to 7MB to fit within macOS limits.